src/Xabe.FFmpeg/FFmpeg.cs

```diff
@@ -143,6 +168,50 @@ private void ValidateExecutables()
             throw new FFmpegNotFoundException(exceptionMessage);
         }
 
+        private bool IsExecutable(string file)
+        {
+            try
+            {
+                using (var fileStream = File.OpenRead(file))
+                {
+                    var magicNumber = new byte[4];
+                    var appMagicNumber = new byte[4];
+                    fileStream.Read(magicNumber, 0, 4);
+                    var provider = new OperatingSystemProvider();
+
+                    switch (provider.GetOperatingSystem())
+                    {
+                        case OperatingSystem.Windows64:
```

Comment:
You can have one code for multiple cases

case OperatinSystem.Windows64:
case OperatingSystem.Windows32:
    // code